sktime/forecasting/arch.py

```diff
@@ -0,0 +1,405 @@
+            tol=self.tol,
+            options=self.options,
+            backcast=self.backcast,
+        )
+        return self
+
+    def _predict(self, fh=None, X=None):
+        """Forecast time series at future horizon.
+
+        Parameters
+        ----------
+        fh : guaranteed to be ForecastingHorizon or None, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+            If not passed in _fit, guaranteed to be passed here
+        params : {np.ndarray, Series}
+            Parameters required to forecast. Must be identical in shape to the
+            parameters computed by fitting the model.
+        start : {int, datetime, Timestamp, str}, optional
+            An integer, datetime or str indicating the first observation to
+            produce the forecast for. Datetimes can only be used with pandas
+            inputs that have a datetime index. Strings must be convertible to
+            a date time, such as in '1945-01-01'.
+        align : str, optional
+            Either 'origin' or 'target'. When set of 'origin', the t-th row of
+            forecasts contains the forecasts for t+1, t+2, ..., t+h. When set to
+            'target', the t-th row contains the 1-step ahead forecast from time t-1,
+            the 2 step from time t-2, ..., and the h-step from time t-h. 'target'
+            simplified computing forecast errors since the realization and h-step
+            forecast are aligned.
+        method : {'analytic', 'simulation', 'bootstrap'}
+            Method to use when producing the forecast. The default is analytic. The
+            method only affects the variance forecast generation. Not all volatility
+            models support all methods. In particular, volatility models that do not
+            evolve in squares such as EGARCH or TARCH do not support the 'analytic'
+            method for horizons > 1.
+        simulations : int
+            Number of simulations to run when computing the forecast using either
+            simulation or bootstrap.
+        rng : callable, optional
+            Custom random number generator to use in simulation-based forecasts. Must
+            produce random samples using the syntax rng(size) where size the 2-element
+            tuple (simulations, horizon).
+        random_state : RandomState, optional
+            NumPy RandomState instance to use when method is 'bootstrap'
+        reindex : bool, optional
+            Whether to reindex the forecasts to have the same dimension as the series
+            being forecast. Prior to 4.18 this was the default. As of 4.19 this is now
+            optional. If not provided, a warning is raised about the future change in
+            the default which will occur after September 2021.
+
+        Returns
+        -------
+        y_pred : pd.Series
+            Point predictions
+        """
+        if fh is not None:
+            self.fh = fh
```

Comment:
I think this is not needed and done in base class, but you might just check that. There is `check_fh()` called I think

Reply:
okay, will replace it with the `check_fit` method.

Reply:
no, kindly do not do anything of that kind at all. Input checks are carried out by the base class.
All the inner methods should implement is model logic.

Reply:
Oh, I got what you meant.

Reply:
what happens if you pass the horizon both in `fit` and `predict`, to `arch`?

Reply:
Very sorry, excuse me. Only predict method takes horizon param. The issue is resolved.